src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -114,6 +114,25 @@ class ListBuffer[A]
     this
   }
 
+  // Overridden for performance
+  override final def addAll(xs: IterableOnce[A]): this.type = {
+    val it = xs.iterator
+    if (it.hasNext) {
+      ensureUnaliased()
+      val last1 = new ::[A](it.next(), Nil)
+      if (len == 0) first = last1 else last0.next = last1
+      last0 = last1
```

Comment:
do we get anything my moving len and last0 to local vars until end of the while loop 
i.e. is it faster to do ….
```
override final def addAll(xs: IterableOnce[A]): this.type = {
    val it = xs.iterator
    if (it.hasNext) {
      ensureUnaliased()
//cache local len and last0
      var len = this.len
      var last0 = this.last0
      val last1 = new ::[A](it.next(), Nil)
      if (len == 0) first = last1 else last0.next = last1
      last0 = last1
      len += 1
      while (it.hasNext) {
        val last1 = new ::[A](it.next(), Nil)
        last0.next = last1
        last0 = last1
        len += 1
      }
//flush local len and last0
      this.len = len
      this.last0 = last0
    }
    this
  }
```

Reply:
Doesn't seem to make a difference to the benchmark.